### ISSUE TITLE
Compile error `#error config.h must be included before this header`

### ISSUE THREAD

**Issue by christian-heusel** [ptesarik/libkdumpfile#86]
When trying to package version `v0.5.5` for Arch Linux I'm getting the following compile error, this is with `binutils` version `2.43+r4+g7999dae6961-1`. I'm not sure if this is an issue with binutils or with `libkdumpfile` 🤔 
```
==> Starting build()...
checking for a BSD-compatible install... /usr/bin/install -c
checking whether sleep supports fractional seconds... yes
checking filesystem timestamp resolution... 0.01
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking xargs -n works... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking if libraries can be versioned... yes
checking for doxygen... no
configure: WARNING: doxygen not found - will not generate any doxygen documentation
checking for perl... /usr/bin/perl
checking size of long... 8
checking size of off_t... 8
checking for mmap64... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for zlib... yes
checking for lzo2... yes
checking for snappy... yes
checking for libzstd... yes
checking for library containing pthread_create... none required
checking for dis-asm.h... yes
checking whether disassembler supports syntax highlighting... yes
checking for disassembler in -lopcodes... yes
checking whether disassembler requires -lbfd... no
checking whether disassembler requires -lsframe... no
checking whether disassembler requires -liberty... no
checking whether disassembler requires -lz... no
checking whether disassembler requires -lzstd... no
checking whether disassembler requires -ldl... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating examples/Makefile
config.status: creating include/Makefile
config.status: creating include/libkdumpfile/Makefile
config.status: creating src/Makefile
config.status: creating src/addrxlat/Makefile
config.status: creating src/kdumpfile/Makefile
config.status: creating python/Makefile
config.status: creating tests/Makefile
config.status: creating tools/Makefile
config.status: creating tools/kdumpid/Makefile
config.status: creating libaddrxlat.pc
config.status: creating libkdumpfile.pc
config.status: creating include/libkdumpfile/kdumpfile.h
config.status: creating include/libkdumpfile/addrxlat.h
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
make  all-recursive
make[1]: Entering directory '/build/libkdumpfile/src/libkdumpfile-0.5.5'
Making all in include
make[2]: Entering directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/include'
Making all in libkdumpfile
make[3]: Entering directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/include/libkdumpfile'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/include/libkdumpfile'
make[3]: Entering directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/include'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/include'
make[2]: Leaving directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/include'
Making all in src
make[2]: Entering directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/src'
Making all in addrxlat
make[3]: Entering directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/src/addrxlat'
[...]
gcc -DHAVE_CONFIG_H -I. -I../..  -I../../include -DVER_MAJOR=1 -DVER_MINOR=7   -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/libkdumpfile/src=/usr/src/debug/libkdumpfile -flto=auto -MT s390.o -MD -MP -MF .deps/s390.Tpo -c -o s390.o s390.c
gcc -DHAVE_CONFIG_H -I. -I../..  -I../../include -DVER_MAJOR=1 -DVER_MINOR=7   -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/libkdumpfile/src=/usr/src/debug/libkdumpfile -flto=auto -MT x86.o -MD -MP -MF .deps/x86.Tpo -c -o x86.o x86.c
In file included from /usr/include/dis-asm.h:36,
                 from ppc64.c:18:
/usr/include/bfd.h:35:2: error: #error config.h must be included before this header
   35 | #error config.h must be included before this header
      |  ^~~~~
In file included from /usr/include/dis-asm.h:36,
                 from x86.c:5:
/usr/include/bfd.h:35:2: error: #error config.h must be included before this header
   35 | #error config.h must be included before this header
      |  ^~~~~
In file included from /usr/include/dis-asm.h:36,
                 from ppc.c:18:
/usr/include/bfd.h:35:2: error: #error config.h must be included before this header
   35 | #error config.h must be included before this header
      |  ^~~~~
In file included from /usr/include/dis-asm.h:36,
                 from s390.c:18:
/usr/include/bfd.h:35:2: error: #error config.h must be included before this header
   35 | #error config.h must be included before this header
      |  ^~~~~
make[3]: *** [Makefile:522: ppc64.o] Error 1
make[3]: *** Waiting for unfinished jobs....
mv -f .deps/search.Tpo .deps/search.Po
make[3]: *** [Makefile:523: x86.o] Error 1
make[3]: *** [Makefile:522: ppc.o] Error 1
make[3]: *** [Makefile:522: s390.o] Error 1
mv -f .deps/util.Tpo .deps/util.Po
mv -f .deps/main.Tpo .deps/main.Po
make[3]: Leaving directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/tools/kdumpid'
make[2]: *** [Makefile:419: all-recursive] Error 1
make[2]: Leaving directory '/build/libkdumpfile/src/libkdumpfile-0.5.5/tools'
make[1]: *** [Makefile:540: all-recursive] Error 1
make[1]: Leaving directory '/build/libkdumpfile/src/libkdumpfile-0.5.5'
make: *** [Makefile:447: all] Error 2
==> ERROR: A failure occurred in build().
```

**Comment by ptesarik** [comment]
Hi Christian,
thank you for your report. I wonder if you are able to build the original [kdumpid](https://github.com/ptesarik/kdumpid) sources separately. If yes, then this might be yet another fallout from the integration. If not, then we'll have to look at how the binutils disassembler should be used on Arch Linux, and it may turn out to be an issue with binutils (or how they are packaged).
If all you need is a new version of the library, you can skip building this tool with `./configure --disable-kdumpid`.

**Comment by ptesarik** [comment]
Oh, I have just seen #84. That explains it. Let me also add the link to the BFD bug here: https://sourceware.org/PR14243

**Comment by ptesarik** [comment]
Fixed by #84.
